packages/wdio-sync/src/index.js

```diff
@@ -1,83 +1,31 @@
-import Fiber from './fibers'
+import deasync from 'deasync'
 
-import executeHooksWithArgs from './executeHooksWithArgs'
-import runFnInFiberContext from './runFnInFiberContext'
-import wrapCommand from './wrapCommand'
-
-import { STACKTRACE_FILTER_FN } from './constants'
-const defaultRetries = { attempts: 0, limit: 0 }
-
-/**
- * execute test or hook synchronously
- *
- * @param  {Function} fn         spec or hook method
- * @param  {Number}   retries    { limit: number, attempts: number }
- * @param  {Array}    args       arguments passed to hook
- * @return {Promise}             that gets resolved once test/hook is done or was retried enough
- */
-const executeSync = async function (fn, retries = defaultRetries, args = []) {
-    /**
-     * User can also use the `@wdio/sync` package directly to run commands
-     * synchronously in standalone mode. In this case we neither have
-     * `global.browser` nor `this`
-     */
-    if (global.browser) {
-        delete global.browser._NOT_FIBER
-    }
-    if (this) {
-        this.wdioRetries = retries.attempts
+export default function waitForPromise (promise) {
+    if (!(promise instanceof Promise) || !process.env.WDIO_SYNC_ENABLED) {
```

Comment:
If string false is passed to wdio sync enabled the condition won't work as expected

Reply:
removed that part